web/src/engine/attachment/src/pageContextAttachment.ts

```diff
@@ -999,8 +999,44 @@ export class PageContextAttachment extends EventEmitter<EventMap> {
     }
 
     for(k = 0; k < inputElementRemovals.length; k++) {
+      let matched = false;
       const elem = inputElementRemovals[k];
-      if(this.isKMWInput(elem)) {
+
+      // Note:  for iframes, .contentWindow has already been deleted by this point!
+      if(elem instanceof elem.ownerDocument.defaultView.HTMLIFrameElement) {
```

Comment:
This change was strongly motivated once I got to writing unit tests.  The natural test one would want to write regarding the `disabled` event, if writing from a TDD perspective, was failing _hard_ because of the problem noted above.  This was leading to significant confusion on my end, as it "smelled like a bug", even though it had a reasonable explanation.

The short version:  if you pull an iframe element out of the page hierarchy, its content document completely disappears.  MutationObservers won't be able to catch it, as the document's removal makes accessing any context-supplying elements impossible.  This meant that we couldn't send any events about the elements having become detached.  Fortunately, our class fields provide enough data to allow a workaround.

It did take a little effort and testing to resolve, but this change in implementation helps us meet a more natural set of expectations, and thus a more intuitive test spec.